verify_release

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python
+
+# Copyright 2022, TUF contributors
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+"""verify_release - verify that published release matches a locally built one
+
+Builds a release from current commit and verifies that the release artifacts
+on GitHub and PyPI match the built release artifacts.
+"""
+
+import json
+import os
+import subprocess
+import sys
+from filecmp import dircmp
+from tempfile import TemporaryDirectory
+
+import requests
+
+# Project variables
+# Note that only these project artifacts are supported:
+# [f"{PYPI_PROJECT}-{VER}-none-any.whl",  f"{PYPI_PROJECT}-{VER}.tar.gz"]
+GITHUB_ORG = "theupdateframework"
+GITHUB_PROJECT = "python-tuf"
+PYPI_PROJECT = "tuf"
+
+
+def build(build_dir: str) -> str:
+    """Build release locally. Return version as string"""
+    cmd = ["python3", "-m", "build", "--outdir", build_dir]
+    subprocess.run(cmd, stdout=subprocess.DEVNULL, check=True)
+    build_version = None
+    for filename in os.listdir(build_dir):
+        prefix, postfix = f"{PYPI_PROJECT}-", ".tar.gz"
+        if filename.startswith(prefix) and filename.endswith(postfix):
+            build_version = filename[len(prefix) : -len(postfix)]
+    assert build_version
+    return build_version
+
+
+def get_git_version() -> str:
+    """Return version string from git describe"""
+    cmd = ["git", "describe"]
+    process = subprocess.run(cmd, text=True, capture_output=True, check=True)
+    assert process.stdout.startswith("v") and process.stdout.endswith("\n")
+    return process.stdout[1:-1]
+
+
+def get_github_version() -> str:
+    """Return version string of latest GitHub release"""
+    release_json = f"https://api.github.com/repos/{GITHUB_ORG}/{GITHUB_PROJECT}/releases/latest"
+    releases = json.loads(requests.get(release_json).content)
+    return releases["tag_name"][1:]
+
+
+def get_pypi_version() -> str:
+    """Return latest version string available on PyPI"""
+    # pip is sad: it can't even tell what the newest available version is...
+    # FIXME: Maybe think of something better than downloading latest tarball here?
```

Comment:
I intentionally wanted to find the version that pip considers current since that's what matters in the end (obviously the two should be the same but...).

I suppose the method name is a bit misleading in that sense

Reply:
Did you consider using experimental `pip index versions tuf` or parsing error output of `pip install tuf==`? (quoting [SO](https://stackoverflow.com/questions/4888027/python-and-pip-list-all-versions-of-a-package-thats-available/26664162#26664162))

Reply:
well, I have not found a way to find the answer to _"what is **current** version according to pip?"_ I think none of those answer this question.

Reply:
Okay, that all makes sense. :)

Reply:
maybe I've left out the reasoning for this: A malicious maintainer could upload tuf version 99. In this case it's not enough that we check that locally built 1.0 matches pypi version 1.0: end users are still getting version 99 if they do `pip install tuf`.

Reply:
I've improved naming and comments, did not modify the code